.github/workflows/java-gradle-artifact.yaml

```
@@ -0,0 +1,171 @@
+name: Reusable workflow for building and publishing java gradle artifacts
+
+on:
+  workflow_call:
+    inputs:
+      java-distribution:
+        description: "Java distribution to be installed. (Default is temurin)"
+        required: false
+        type: string
+        default: "temurin"
+      java-version:
+        description: "Java version to be installed. (Default is 11)"
+        required: false
+        type: string
+        default: "11"
+      gradle-version:
+        description: "Gradle version to be installed. (Default is 6.7.1)"
+        required: false
+        type: string
+        default: "6.7.1"
+      gradle-cache:
+        description: "Gradle cache to use. Options are gradle, maven, sbt. (Default is gradle)"
+        required: false
+        type: string
+        default: "gradle"
+      integration-tests:
+        description: "If true, the action is executing integration tests (Default is true)"
+        required: false
+        type: boolean
+        default: true
+      docker-publisher:
+        description: "Publisher to prefix docker image."
```

```suggestion
        description: "Publisher to prefix Docker image."
```

Nit, but casing of Docker, Gradle, Java etc. is inconsistent